### PR TITLE
Remove when library

### PR DESCRIPTION
**Pull request by rich-nguyen**
## What does this change?
We rely on polyfills to provide `Promise`, and we can't import `Promise` in `src/javascripts`. This removes `Promise` from `src/javascripts-legacy`. 

## Tested in CODE?
Yes

### CODE REVIEW

**Review by SiAdcock — APPROVED** [review]
❤️  it

File: static/test/javascripts-legacy/spec/commercial/article-body-adverts.spec.js [hunk 1]
```diff
@@ -134,10 +134,12 @@ define([
             });
 
             it('inserts up to ten adverts when DFP returns empty merchandising components', function (done) {
-                spaceFillerStub.onCall(0).returns(Promise.resolve(false));
-                spaceFillerStub.onCall(1).returns(Promise.resolve(0));
-                spaceFillerStub.onCall(2).returns(Promise.resolve(2));
-                spaceFillerStub.onCall(3).returns(Promise.resolve(8));
+                // The 0 is for addInlineMerchAd, failing to add a merchandising component.
+                spaceFillerStub.onCall(0).returns(Promise.resolve(0));
```
Comment (rich-nguyen) [inline static/test/javascripts-legacy/spec/commercial/article-body-adverts.spec.js]: I did the same as you @JonNorman

**Review by sndrs — APPROVED** [review]
📅  now

**Comment by rich-nguyen** [comment]
@sndrs I love a PR revival.

**Comment by sndrs** [comment]
i've got a half a branch that updates to lodash 4 if you want that too 😉

**Pull request merged**